src/common/AudioStream.cpp

```diff
@@ -166,6 +166,12 @@ ResultWithValue<int32_t> AudioStream::waitForAvailableFrames(int32_t numFrames,
     if (numFrames == 0) return Result::OK;
     if (numFrames < 0) return Result::ErrorOutOfRange;
 
+    // Make sure we don't try to wait for more frames than the buffer can hold.
+    // Subtract framesPerBurst because this is often called from a callback
+    // and we don't want to be sleeping if the buffer is close to overflowing.
+    const int32_t maxAvailableFrames = getBufferCapacityInFrames() - getFramesPerBurst();
```

Comment:
Is there ever a case where getFramesPerBurst is equal to getBufferCapacityInFrames?

Reply:
In general the capacity would always be at least one burst.
But now that you mention it the code seems unsafe. I will clip against zero or one burst.